pkg/storaged/btrfs/subvolume.jsx

```diff
@@ -44,6 +47,107 @@ function subvolume_mount(volume, subvol, forced_options) {
     mounting_dialog(client, block, "mount", forced_options, subvol);
 }
 
+function get_mount_point_in_parent(volume, subvol) {
+    const block = client.blocks[volume.path];
+    const subvols = client.uuids_btrfs_subvols[volume.data.uuid];
+    if (!subvols)
+        return null;
+
+    for (const p of subvols) {
+        if ((p.pathname == "/" || (subvol.pathname.substring(0, p.pathname.length) == p.pathname &&
+                                           subvol.pathname[p.pathname.length] == "/")) &&
+                    is_mounted(client, block, p)) {
+            const [, pmp] = get_fstab_config_with_client(client, block, false, p);
+            if (p.pathname == "/")
+                return pmp + "/" + subvol.pathname;
+            else
+                return pmp + subvol.pathname.substring(p.pathname.length);
+        }
+    }
+    return null;
+}
+
+function set_mount_options(subvol, block, vals) {
+    const mount_options = [];
+
+    if (vals.mount_options.ro)
+        mount_options.push("ro");
+    if (vals.at_boot == "never")
+        mount_options.push("x-cockpit-never-auto");
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19849-20240117-090134-2dfecb5e-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)